app/models/concerns/orchestration/compute.rb

```diff
@@ -11,7 +11,7 @@ module Orchestration::Compute
   end
 
   def compute?
-    compute_resource_id.present? && (compute_attributes.present? || uuid.present?)
+    compute_resource_id.present? && compute_resource.present? && (compute_attributes.present? || uuid.present?)
```

Comment:
The shoulda matchers test inserts an arbitrary id in the `compute_resource_id` column. This leads to the condition, that `compute_resource_id` is present but `compute_resource` is not. I guess, `compute_resource_id.present?` is there instead of `compute_resource.present?` because it's faster. I thought, we could keep it but add the more specific check to catch the described errors.